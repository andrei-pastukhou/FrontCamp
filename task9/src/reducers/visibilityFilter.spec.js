import reducer from './visibilityFilter'
const initialState = ''

describe('visibilityFilter reducer', () => {
    it('should handle initial state', () => {
        expect(
            reducer(undefined, {})
        ).toEqual(initialState)
    });
    it('should handle FILTER_POST', () => {
        expect(
            reducer(initialState, {
                type: 'FILTER_POST',
                author: 'testAuthor'
            })
        ).toEqual('testAuthor')
    });
});

describe('visibilityFilter reduxer with snapshots ', () => {
    it('should handle initial state', () => {
        const action = {}
        expect(reducer(undefined, action)).toMatchSnapshot();
    });
    it('should handle FILTER_POST', () => {
        const action = {
            type: 'FILTER_POST',
            author: 'testAuthor'
        }
        expect(reducer(initialState, action)).toMatchSnapshot();
    });
});
