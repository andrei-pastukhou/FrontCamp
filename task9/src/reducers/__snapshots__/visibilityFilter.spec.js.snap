// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`visibilityFilter reduxer with snapshots  should handle FILTER_POST 1`] = `"testAuthor"`;

exports[`visibilityFilter reduxer with snapshots  should handle initial state 1`] = `""`;
